front-end-tools/src/components/InitComponent.tsx

```diff
@@ -95,11 +91,14 @@ export default function InitComponent(props: ConnectionProps) {
     const [moduleReferenceLengthError, setModuleReferenceLengthError] = useState<string | undefined>(undefined);
     const [schemaError, setSchemaError] = useState<string | undefined>(undefined);
 
-    const [isModuleReferenceAlreadyDeployedStep2, setIsModuleReferenceAlreadyDeployedStep2] = useState(false);
+    const [isModuleReferenceAlreadyDeployed, setIsModuleReferenceAlreadyDeployed] = useState(false);
 
     const [txHash, setTxHash] = useState<string | undefined>(undefined);
 
-    const [uploadedModuleSchemaBase64, setUploadedModuleSchemaBase64] = useState<string | undefined>(undefined);
+    const [schema, setSchema] = useState<string | undefined>(undefined);
+    const [moduleReference, setModuleReference] = useState<ModuleReference.Type | undefined>(
```

Comment:
In case the user selects `deriveFromStep1` but the module reference is not set in step 1, we set the module reference back to `undefined` to catch that case. 
https://github.com/Concordium/concordium-smart-contract-tools/pull/154/files#diff-29a4ad93b8ac538e67ec59665a4623b87c298a4fc584824c3dc37430b233fe6dR327